modules/standard/IO.chpl

```diff
@@ -4022,6 +4040,83 @@ proc channel.writebits(v:integral, nbits:integral):bool throws {
   return try this.write(new ioBits(v:uint(64), nbits:int(8)));
 }
 
+proc channel.writeBinary(arg:numeric, param endian:ioendian = ioendian.native) throws {
```

Comment:
The new functions seem to be missing docs strings & we should put forward the prototypes & docs strings for broader API review.

Reply:
Also, I don't see anything about writeBinary for strings, which is a capability that the format strings previously supported. (But I don't know if people were using it). I think discussion of the API for these functions should also include asking if we want these functions or related functions to work with strings or bytes. Could you make an issue that is focused on the API design for the new functions here?

Reply:
(To be clear, it would be OK with me to merge this PR while that discussion is ongoing, but I think we should at least start it.)

Reply:
I probably should have done this as a draft PR, but I wanted to get feedback on what I have so far before figuring out how docs strings work. I wasn't sure what the plan was for strings, and if it should be part of this PR or a separate PR.

Reply:
Separate PR is fine but we need to get that design discussion going, somewhere.